saleor/graphql/checkout/mutations/checkout_lines_update.py

```diff
@@ -162,10 +160,47 @@ def clean_input(
             checkout_lines_data,
             checkout_info,
             lines_info,
-            manager,
-            replace,
         )
 
+    @classmethod
+    def process_lines_input(
+        cls,
+        info,
+        checkout,
+        variants,
+        checkout_lines_data,
+        checkout_info,
+        replace=False,
+        raise_error_for_missing_lines=False,
+    ):
+        try:
+            return super().process_lines_input(
+                info,
+                checkout,
+                variants,
+                checkout_lines_data,
+                checkout_info,
+                replace=True,
+                # set to true, as during the update we want to be sure that any deleted line
+                # in the meantime will raise an exception instead of creating the new line.
+                raise_error_for_missing_lines=True,
+            )
+        except NonExistingCheckoutLines as e:
```

Comment:
Maybe I'm missing something, but why do we have error handling in `checkoutLinesUpdate`, but not in the 
 `checkoutLinesAdd`?

Reply:
@maarcingebala  because the flow is different in both cases. I mean:
- when calling `add`, you expect to add the variant to the checkout.  You don't relay in the input on `CheckoutLine.id`, but on the provided `variant.id`, so as a result you want to have a line with the variant from the input.  No matter if the `CheckoutLine` with `variant.id` is present or not, you want to have a single line with this `variant.id`
- but when calling `update`, you provide specific `CheckoutLine.id` that you want to update. If the line is not present as it was deleted in the meantime, we raise the same exception as we did when validating the input.